internal/elasticsearch/transform/transform.go

```diff
@@ -0,0 +1,843 @@
+			return nil, err
+		}
+		transform.Latest = latest
+	}
+
+	if v, ok := d.GetOk("frequency"); ok && isSettingAllowed(ctx, "frequency", serverVersion) {
+		transform.Frequency = v.(string)
+	}
+
+	if v, ok := d.GetOk("metadata"); ok && isSettingAllowed(ctx, "metadata", serverVersion) {
+		var metadata map[string]interface{}
+		if err := json.NewDecoder(strings.NewReader(v.(string))).Decode(&metadata); err != nil {
+			return nil, err
+		}
+		transform.Meta = metadata
+	}
+
+	if v, ok := d.GetOk("retention_policy"); ok && v != nil && isSettingAllowed(ctx, "retention_policy", serverVersion) {
+		definedRetentionPolicy := v.([]interface{})[0].(map[string]interface{})
+
+		if v, ok := definedRetentionPolicy["time"]; ok {
+			retentionTime := models.TransformRetentionPolicyTime{}
+			var definedRetentionTime = v.([]interface{})[0].(map[string]interface{})
+			if f, ok := definedRetentionTime["field"]; ok {
+				retentionTime.Field = f.(string)
+			}
+			if ma, ok := definedRetentionTime["max_age"]; ok {
+				retentionTime.MaxAge = ma.(string)
+			}
+			transform.RetentionPolicy = &models.TransformRetentionPolicy{
+				Time: retentionTime,
+			}
+		}
+	}
+
+	if v, ok := d.GetOk("sync"); ok {
+		definedSync := v.([]interface{})[0].(map[string]interface{})
+
+		if v, ok := definedSync["time"]; ok {
+			syncTime := models.TransformSyncTime{}
+			var definedSyncTime = v.([]interface{})[0].(map[string]interface{})
+			if f, ok := definedSyncTime["field"]; ok {
+				syncTime.Field = f.(string)
+			}
+			if d, ok := definedSyncTime["delay"]; ok {
+				syncTime.Delay = d.(string)
+			}
+			transform.Sync = &models.TransformSync{
+				Time: syncTime,
+			}
+		}
+	}
+
+	// settings
+	settings := models.TransformSettings{}
+	setSettings := false
+
+	if v, ok := d.GetOk("align_checkpoints"); ok && isSettingAllowed(ctx, "align_checkpoints", serverVersion) {
+		setSettings = true
+		ac := v.(bool)
+		settings.AlignCheckpoints = &ac
+	}
+	if v, ok := d.GetOk("dates_as_epoch_millis"); ok && isSettingAllowed(ctx, "dates_as_epoch_millis", serverVersion) {
+		setSettings = true
+		dem := v.(bool)
+		settings.DatesAsEpochMillis = &dem
+	}
+	if v, ok := d.GetOk("deduce_mappings"); ok && isSettingAllowed(ctx, "deduce_mappings", serverVersion) {
+		setSettings = true
+		dm := v.(bool)
+		settings.DeduceMappings = &dm
+	}
+	if v, ok := d.GetOk("docs_per_second"); ok && v.(float64) >= 0 && isSettingAllowed(ctx, "docs_per_second", serverVersion) {
+		setSettings = true
+		dps := v.(float64)
+		settings.DocsPerSecond = &dps
+	}
+	if v, ok := d.GetOk("max_page_search_size"); ok && v.(int) > 10 && isSettingAllowed(ctx, "max_page_search_size", serverVersion) {
+		setSettings = true
+		mpss := v.(int)
+		settings.MaxPageSearchSize = &mpss
+	}
+	if v, ok := d.GetOk("num_failure_retries"); ok && v.(int) >= -1 && isSettingAllowed(ctx, "num_failure_retries", serverVersion) {
+		setSettings = true
+		nfr := v.(int)
+		settings.NumFailureRetries = &nfr
+	}
```

Comment:
Are we able to remove the `v.(int) > 10` style conditions in these? Terraform should be validating these options before executing the provider